include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -524,43 +541,92 @@ struct __usm_host_or_buffer_storage
 #endif
     }
 
+    inline bool
+    __use_USM_allocations(sycl::queue __queue)
+    {
+        auto __device = __queue.get_device();
+        if (!__device.has(sycl::aspect::usm_device_allocations))
+            return false;
+        return true;
+    }
```

Comment:
We might need to guard this feature similarly to `_ONEDPL_SYCL_USM_HOST_PRESENT` in case the underlying compiler issue is present for all types of USM memory. @AidanBeltonS please double-check this implementation with older release compilers.

Reply:
I have checked and added the macro to guard the USM approach to maintain backward compatibility